packages/api-plugin-promotions/src/queries/promotions.js

```diff
@@ -8,21 +8,27 @@
 export default async function promotions(context, shopId, filter) {
   const { collections: { Promotions } } = context;
 
-  const selector = { shopId };
+  const selector = { shopId, state: { $ne: "archived" } };
 
   if (filter) {
-    const { enabled, startDate, endDate } = filter;
+    const { enabled, startDate, endDate, state } = filter;
     // because enabled could be false we need to check for undefined
     if (typeof enabled !== "undefined") {
       selector.enabled = enabled;
     }
+
+    if (state || (state === "archived" && await context.userHasPermission("reaction:legacy:promotions", "read:archived", { shopId }))) {
+      selector.state = { $eq: state };
+    }
+
```

Comment:
if state === "archived" the permission check will not be reached because it pass the first check.
Or can we just throw an error?

```suggestion
    if (state ) {
      const canUpdate = state !== "archived" || await context.userHasPermission("reaction:legacy:promotions", "read:archived", { shopId }));
      if (canUpdate) selector.state = { $eq: state };
    }
```